src/EditorFeatures/Core.Wpf/InlineRename/UI/SmartRename/SmartRenameViewModel.cs

```diff
@@ -163,43 +164,53 @@ private void FetchSuggestions(bool isAutomaticOnInitialization)
 
     private async Task GetSuggestionsTaskAsync(bool isAutomaticOnInitialization, CancellationToken cancellationToken)
     {
-        if (isAutomaticOnInitialization)
+        try
         {
-            await Task.Delay(_smartRenameSession.AutomaticFetchDelay, cancellationToken)
-                .ConfigureAwait(false);
-        }
+            this.IsInPreparation = true;
```

Comment:
i would like some sort of threading asserts here.  can talk mroe on what we often do in roslyn.

Reply:
we should also assert that we're not already 'in preparation'.

Reply:
please elaborate

Reply:
> This is already indirectly asserted above, through

I mean a literal retail assert that IsInPreparation must be false before setting it to true.  Alternative, this can be in the setter, which retail asserts that the value is always flipping.

Reply:
> Can you elaborate on threading asserts? This method should be free threaded;

It is not ok to be free threaded.  two threads might call in here, and end up with one setting the boolean back to false, while the other is still running.

Reply:
Makes sense, I added these:
```
        _threadingContext.ThrowIfNotOnUIThread();
        RoslynDebug.Assert(!this.IsInPreparation);
```